src/formatDate.js

```diff
@@ -50,7 +50,55 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateDigits = date.match(/\d+/g);
+  const signedDates = {};
+  const arrangedDates = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    signedDates[fromFormat[i]] = dateDigits[i];
+  }
+
+  if (fromFormat.includes('YY')
+    && toFormat.includes('YYYY')
+    && signedDates.YY < 30) {
+    signedDates.YYYY = '20' + signedDates.YY;
+  }
+
+  if (fromFormat.includes('YY')
+    && toFormat.includes('YYYY')
+    && signedDates.YY >= 30) {
+    signedDates.YYYY = '19' + signedDates.YY;
+  }
+
+  if (fromFormat.includes('YYYY')
+    && toFormat.includes('YY')) {
+    signedDates.YY = signedDates.YYYY.slice(2);
+  }
```

Comment:
I am sure you can make it without such long and repetitive conditions in your ifs